From 37880404e8fc0ce656510fd7d4ca0d1e3d181b0f Mon Sep 17 00:00:00 2001
From: Sachin Prabhu <sprabhu@redhat.com>
Date: Mon, 28 Jul 2014 12:16:59 -0400
Subject: [fs] cifs: Do not send ClientGUID on SMB2.02 dialect

Message-id: <1406549822-10376-67-git-send-email-sprabhu@redhat.com>
Patchwork-id: 86768
O-Subject: [RHEL7.1 PATCH 66/69] [bz1082049]Do not send ClientGUID on SMB2.02 dialect
Bugzilla: 1082049
RH-Acked-by: J. Bruce Fields <bfields@redhat.com>
RH-Acked-by: Steve Dickson <SteveD@redhat.com>

From: Steve French <smfrench@gmail.com>

ClientGUID must be zero for SMB2.02 dialect.  See section 2.2.3
of MS-SMB2. For SMB2.1 and later it must be non-zero.

Signed-off-by: Steve French <smfrench@gmail.com>
CC: Sachin Prabhu <sprabhu@redhat.com>
(cherry picked from commit 3c5f9be108783c05cade918d29c8711b236acb1d)
Signed-off-by: Sachin Prabhu <sprabhu@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/fs/cifs/smb2pdu.c b/fs/cifs/smb2pdu.c
index dc44610..0feb743 100644
--- a/fs/cifs/smb2pdu.c
+++ b/fs/cifs/smb2pdu.c
@@ -375,7 +375,12 @@ SMB2_negotiate(const unsigned int xid, struct cifs_ses *ses)
 
  req->Capabilities = cpu_to_le32(ses->server->vals->req_capabilities);
 
- memcpy(req->ClientGUID, server->client_guid, SMB2_CLIENT_GUID_SIZE);
+ /* ClientGUID must be zero for SMB2.02 dialect */
+ if (ses->server->vals->protocol_id == SMB20_PROT_ID)
+  memset(req->ClientGUID, 0, SMB2_CLIENT_GUID_SIZE);
+ else
+  memcpy(req->ClientGUID, server->client_guid,
+   SMB2_CLIENT_GUID_SIZE);
 
  iov[0].iov_base = (char *)req;
  /* 4 for rfc1002 length field */
-- 
1.7.1